src/screens/lobby/lobby.jsx

```diff
@@ -4,38 +4,86 @@ import LeaveGameModal from '../../components/modals/leave-game/leave-game';
 import SelectCharacterModal from '../../components/modals/select-character/select-character';
 import Header from '../../components/header/header';
 import ScreenWrapper from '../../components/wrappers/screen-wrapper/screen-wrapper';
-import { useState } from 'react';
+import { useContext, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { PROCESSING_QUESTION } from '../../constants/constants';
 import './lobby.scss';
+import GameDataContext from '../../contexts/game-data-context';
+import { findGameById } from '../../services/games-service';
 
 function Lobby() {
+  const { gameData, setGameData, playerId } = useContext(GameDataContext);
   const [showLeaveModal, setLeaveModalActive] = useState(false);
   const [showSuggestModal, setSuggestModal] = useState(false);
+  const [suggestBtn, setSuggestBtn] = useState(true);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (!gameData.data.status) navigate('/');
+  });
+
+  useEffect(() => {
+    const checkStatus = setTimeout(async () => {
+      setGameData(await findGameById(playerId, gameData.data.id));
+    }, 1000);
+
+    return () => clearTimeout(checkStatus);
+  });
+
+  useEffect(() => {
+    if (gameData.data.status === PROCESSING_QUESTION) {
+      navigate('/play');
+    }
+  }, [gameData, navigate]);
+
+  const players = gameData.data.players;
+  const currentPlayer =
+    players && players.find((player) => player.player.name === playerId);
+  const playersWithoutYou =
+    players && players.filter((player) => player.player.name !== playerId);
 
   return (
     <ScreenWrapper>
       <div className="input-screen">
         <Header inLobby />
         <div className="input-screen__player">
           <div className="input-screen__player-card-wrapper">
-            {[1, 2, 3, 4].map((num) => (
+            {currentPlayer && (
               <PlayerCard
-                key={num}
-                avatarClassName={`avatar0${num}`}
-                name={`Player ${num}`}
-                playerStausClassName={'no'}
+                avatarClassName={`avatar0${1}`}
+                name={`Player ${1}`}
+                playerStatusClassName={
+                  currentPlayer.state === 'NOT_READY' ? 'no' : 'yes'
+                }
                 isYou
               />
-            ))}
+            )}
+            {playersWithoutYou ? (
+              playersWithoutYou.map((player, index) => (
+                <PlayerCard
+                  key={player.player.name}
+                  avatarClassName={`avatar0${index + 2}`}
+                  name={`Player ${index + 2}`}
```

Comment:
maybe `player.name` ?

Reply:
maybe, but now it's just an id and looks not good. For now we have not possibility to get normal name. So should I replace this with player.name anyway?

Reply:
ok, skip for now